### PR TITLE
Update default backup filename to id-name-softversion-date-time

### PR DESCRIPTION
**Pull request by bobjacobsen**
Changes the default filename when doing a backup from the CDI pane.  The new form is id-name-softwareversion-date-time, e.g.

```
config.02.01.57.00.04.8B-East_test_board-rev-C7b-2023-02-24-10:06:21.txt
```

### CODE REVIEW

**Comment by bobjacobsen** [comment]
See discussion at https://groups.io/g/layoutcommandcontrol/topic/97184619

File: src/org/openlcb/cdi/swing/CdiPanel.java [hunk 1]
```diff
@@ -422,11 +424,33 @@ public void runBackup() {
...
+        if (!nodeName.isEmpty()) {
+            fileName += "-"+nodeName;
+        }
+        if (rep.getCdiRep() != null && rep.getCdiRep().getIdentification() != null) {
+            // info not present if configuration hasn't been read yet
+            fileName += "-"+rep.getCdiRep().getIdentification().getSoftwareVersion();
+        }
+        fileName += "-"+java.time.LocalDate.now();
+        fileName  += "-"+java.time.LocalTime.now().format( // use default time zone
+                        java.time.format.DateTimeFormatter.ofPattern("HH:mm:ss")
```
Comment (balazsracz) [inline src/org/openlcb/cdi/swing/CdiPanel.java:438]: I think a colon could be problematic in filenames as well (at least under windows). Using dots or just writing it together might be better.

File: src/org/openlcb/cdi/swing/CdiPanel.java [hunk 2]
```diff
@@ -422,11 +424,33 @@ public void runBackup() {
...
+    }
+
+
     public void runRestore() {
         // First select a file to save to.
         fci.setDialogTitle("Open configuration restore file");
         fci.rescanCurrentDirectory();
-        fci.setSelectedFile(new File("config." + rep.getRemoteNodeAsString() + ".txt"));
+
+        fci.setSelectedFile(new File(generateFileName()));
```
Comment (balazsracz) [inline src/org/openlcb/cdi/swing/CdiPanel.java:452]: This surely won't work with the date and time now being part of the filename.
Can the file chooser get something like a glob pattern?

Comment (bobjacobsen) [reply src/org/openlcb/cdi/swing/CdiPanel.java:452]: You're right that this won't end up selecting a file. This goes to the saved directory (if any), but doesn't select a file.

Unfortunately, no, there's no glob capability in the JFileChooser.  Coding one is possible, but it's a non-trivial amount of work.

Another approach might be to present the last filename used by a Save (in the same session) and select that.

**Comment by balazsracz** [comment]
On Wed, Mar 1, 2023 at 5:36 AM Bob Jacobsen ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/org/openlcb/cdi/swing/CdiPanel.java
> <https://github.com/openlcb/OpenLCB_Java/pull/214#discussion_r1121740613>:
>
> >      public void runRestore() {
>          // First select a file to save to.
>          fci.setDialogTitle("Open configuration restore file");
>          fci.rescanCurrentDirectory();
> -        fci.setSelectedFile(new File("config." + rep.getRemoteNodeAsString() + ".txt"));
> +
> +        fci.setSelectedFile(new File(generateFileName()));
>
> You're right that this won't end up selecting a file. This goes to the
> saved directory (if any), but doesn't select a file.
>
> Unfortunately, no, there's no glob capability in the JFileChooser. Coding
> one is possible, but it's a non-trivial amount of work.
>
> Another approach might be to present the last filename used by a Save (in
> the same session) and select that.
>
This sounds like an ok behavior. If there wasn't anything in the current
session, then just show the directory with *.txt in the chooser.

**Comment by bobjacobsen** [comment]
Thanks. Changes made.

**Review by balazsracz — APPROVED** [review]

**Pull request merged**